### PR TITLE
feat: add constants to match spec

### PR DESCRIPTION
**Pull request by davhdavh**
Some constants are missing from the spec in this section: https://openid.net/specs/openid-connect-core-1_0.html#ClaimsParameter

### CODE REVIEW

File: src/OpenIddict.Abstractions/OpenIddictConstants.cs [hunk 1]
```diff
@@ -365,6 +365,15 @@ public static class Parameters
         public const string VerificationUriComplete = "verification_uri_complete";
     }
 
+    public static class ClaimsParameters
```
Comment (kevinchalet) [inline src/OpenIddict.Abstractions/OpenIddictConstants.cs]: `ClaimRequestMembers` may be a better name.

File: src/OpenIddict.Abstractions/OpenIddictConstants.cs [hunk 2]
```diff
@@ -365,6 +365,15 @@ public static class Parameters
...
+    {
+       public const string UserInfo = "userinfo";
```
Comment (kevinchalet) [inline src/OpenIddict.Abstractions/OpenIddictConstants.cs]: Let's name the constant `Userinfo` without a capital I.

**Review by kevinchalet — COMMENTED** [review]
Thanks for your PR.

Please move the new class to the right place so the alphabetical order is respected (same thing for the constants inside the class).

File: src/OpenIddict.Abstractions/OpenIddictConstants.cs (hunk 1 above)
Comment (davhdavh) [reply src/OpenIddict.Abstractions/OpenIddictConstants.cs]: Considering they are called `ClaimParameter` in the spec, I went with that name.
I don't think `ClaimRequestMembers` is a good match though, perhaps `ClaimRequestParameters` or `RequestClaimParameters` or `ClaimRequestParametersMembers`? I still think the `ClaimsParameters` is the best though.

Comment (davhdavh) [reply src/OpenIddict.Abstractions/OpenIddictConstants.cs]: alternatively, a split design:
empty `ClaimRequest` and `ClaimRequest.RequestTypes` with `userinfo` and `id_token`, and `ClaimRequest.Members` with the rest.

Comment (kevinchalet) [reply src/OpenIddict.Abstractions/OpenIddictConstants.cs]: > Considering they are called `ClaimParameter` in the spec, I went with that name.

I'm quite familiar with the spec but I re-read it just to be sure and I don't think it's true. The only parameter mentioned in the 5.5 section is named `claims`: it's the JSON object that is UTF8+formurl-encoded and contains two top-level `id_token` and `userinfo` nodes. These nodes present in the claims request are simply called "members":

image

Comment (davhdavh) [reply src/OpenIddict.Abstractions/OpenIddictConstants.cs]: The url is `openid-connect-core-1_0.html#ClaimsParameter`.
In any case, what is your final preferred naming scheme and I shall update :)

Comment (kevinchalet) [reply src/OpenIddict.Abstractions/OpenIddictConstants.cs]: > The url is `openid-connect-core-1_0.html#ClaimsParameter`. In any case, what is your final preferred naming scheme and I shall update :)

Well, the anchor points to the section title, that clearly says "`claims` Request Parameter": only `claims` is technically defined as a parameter, not the JSON nodes defined in the section itself.

Let's go with `ClaimRequestMembers`. It's not a great name, but it's the most accurate one.

Comment (davhdavh) [reply src/OpenIddict.Abstractions/OpenIddictConstants.cs]: Ok. I felt if you wanted to very strictly keep to the nomenclature used in the actual text of the spec, that we should also split it into `Individual` and `TopLevel`

File: src/OpenIddict.Abstractions/OpenIddictConstants.cs [hunk 3]
```diff
@@ -4,6 +4,8 @@
  * the license and the contributors participating to this project.
  */
 
+using static System.Net.WebRequestMethods;
```
Comment (kevinchalet) [inline src/OpenIddict.Abstractions/OpenIddictConstants.cs]: Looks like an unwanted change (VS can be very annoying with that...)

File: src/OpenIddict.Abstractions/OpenIddictConstants.cs (hunk 1 above)
Comment (kevinchalet) [reply src/OpenIddict.Abstractions/OpenIddictConstants.cs]: Well, it's not wrong, but I'm not so sure it's worth being that specific (it makes names longer). Let's just flatten the class (please also remove the comments, the other classes don't have them).

**Comment by kevinchalet** [comment]
Merged, thanks for your contribution 👏🏻

**Pull request merged**